lib/active_record_query_parts.rb

```diff
@@ -18,42 +18,7 @@ def self.shared_row_lock
     " FOR SHARE"
   end
 
-  # Links for the various string length functions:
-  #   http://www.postgresql.org/docs/9.1/static/functions-binarystring.html
-  #   http://www.postgresql.org/docs/9.1/static/functions-string.html
-  #   http://blog.sqlauthority.com/2007/06/20/sql-server-find-length-of-text-field/
-
-  def self.string_length
-    "LENGTH"
-  end
-
-  def self.text_length
-    "LENGTH"
-  end
-
-  def self.binary_length
-    "LENGTH"
-  end
-
-  def self.concat(*args)
-    args.join('||')
-  end
-
-  def self.regexp(field, regex, qualifier = nil)
-    operator = "SIMILAR TO"
-
-    operator = "NOT #{operator}" if qualifier == :negate
-
-    "#{field} #{operator} '#{regex}'"
-  end
-
-  def self.not_regexp(field, regex)
-    regexp(field, regex, :negate)
-  end
-
   def self.glob_to_sql_like(text)
-    text.tr!('*', '%')
-    text.tr!('?', '_')
-    text
+    text.tr!('*?', '%_') || text
```

Comment:
I'd prefer the two-line spelling here; no need to reason about a pointless branch